spec/cookbook_spec.rb

```diff
@@ -170,13 +170,26 @@
       :result => [
       ],
     },
+    {
+      :name => 'match deepest subfolder first',
+      :files => [
+        {
+          :status => :modified,
+          :path => 'cookbooks/core/cb_one/metadata.rb',
+        },
+      ],
```

Comment:
Without fix this test fails with error:
`Diff:
       @@ -1,2 +1,2 @@
       -[["cb_one", :modified]]
       +[["core", :modified]]
     # ./spec/cookbook_spec.rb:194:in `block (3 levels) in <top (required)>'`
So 'core' is being returned as cookbook name instead of 'cb_one'